content/200-concepts/100-components/02-prisma-client/052-working-with-generated-types/index.mdx

```diff
@@ -0,0 +1,198 @@
+---
+title: 'Advanced usage of generated types'
+metaTitle: 'Advanced usage of generated TypeScript types (Reference)'
+metaDescription: 'Prisma Client provides full type safety for queries, even for partial queries or included relations. This page explains how to leverage the generated types.'
+---
+
+<TopBlock>
+
+The generated code for Prisma Client contains several helpful types that you can use to make your application more type-safe. This page describes patterns for leveraging some of the generated types.
+
+By importing the `Prisma` namespace from `@prisma/client` you can access types related to your defined schema models. These types can be used to compose queries outside of the context of the query, meaning you can re-use query parameters
+or create functions from them.
+
+> Note that you can find all type definitions inside the `index.d.ts` file inside the [generated `.prisma/client` folder](working-with-prismaclient/generating-prisma-client#the-prismaclient-npm-package).
+
+</TopBlock>
+
```

Comment:
Maybe a "accessing custom types" h2 here to generally introduce how to import and access generated types, and a recommendation to use Prisma validator to make sure objects are constructed properly?

```ts
import { PrismaClient, Prisma } from '@prisma/client'

// Do stuff with Prisma.Bla types
```

Reply:
I kind of do that in the top bit. I can move it down, reword it and add an example

Reply:
An example really early on would be 💯